service/src/main/java/uk/gov/hmcts/reform/fpl/service/document/ManageDocumentService.java

```diff
@@ -1157,4 +1163,86 @@ public PlacementEventData updatePlacementNoticesAdmin(CaseData caseData) {
+        Map<DocumentType, List<Element<NotifyDocumentUploaded>>> newDocumentsCTSC = new HashMap<>();
+
+        Map<ConfidentialLevel, Map<DocumentType, List<Element<NotifyDocumentUploaded>>>> resultMapByConfidentialLevel =
+            Map.of(ConfidentialLevel.NON_CONFIDENTIAL, newDocuments,
+                ConfidentialLevel.LA, newDocumentsLA,
+                ConfidentialLevel.CTSC, newDocumentsCTSC);
+
+        for (DocumentType documentType : DocumentType.values()) {
+            for (ConfidentialLevel confidentialLevel : resultMapByConfidentialLevel.keySet()) {
+                if (documentType.getBaseFieldNameResolver() != null) {
+                    String fieldName = documentType.getBaseFieldNameResolver().apply(confidentialLevel);
+
+                    Map<DocumentType, List<Element<NotifyDocumentUploaded>>> newDocMap =
+                            resultMapByConfidentialLevel.get(confidentialLevel);
+
+                    List documentList;
+                    List documentListBefore;
+                    try {
+                        documentList = Optional.ofNullable(ObjectHelper
+                            .getFieldValue(caseData, fieldName, List.class)).orElse(List.of());
+                        documentListBefore = Optional.ofNullable(ObjectHelper
+                            .getFieldValue(caseDataBefore, fieldName, List.class)).orElse(List.of());
+                    } catch (NoSuchMethodException e) {
+                        log.error("No getter for {}. Check configuration of document type: {}",
+                            fieldName, documentType);
+                        continue;
+                    }
+
+                    if (DocumentType.COURT_BUNDLE.equals(documentType)) {
+                        documentList = ((List<Element<HearingCourtBundle>>) documentList).stream()
+                            .map(Element::getValue)
+                            .map(HearingCourtBundle::getCourtBundle)
+                            .flatMap(List::stream)
+                            .toList();
+
+                        documentListBefore = ((List<Element<HearingCourtBundle>>) documentListBefore).stream()
+                            .map(Element::getValue)
+                            .map(HearingCourtBundle::getCourtBundle)
+                            .flatMap(List::stream)
+                            .toList();
+                    }
+
+                    for (Object document : documentList) {
+                        if (!documentListBefore.contains(document)) {
+                            List<Element<NotifyDocumentUploaded>> docList =
+                                Optional.ofNullable(newDocMap.get(documentType))
+                                    .orElse(new ArrayList<>());
+                            newDocMap.putIfAbsent(documentType, docList);
+
+                            docList.add((Element<NotifyDocumentUploaded>) document);
```

Comment:
potentially it causes ClassCastException? worth to try and catch and log an error message in case there is a misconfiguration.